sno/dataset2.py

```diff
@@ -401,9 +395,11 @@ def apply_meta_diff(self, meta_diff, tree_builder):
         if not meta_diff:
             return
 
-        conflicts = False
+        # Applying diffs works even if there is no tree yet created for the dataset,
+        # as is the case when the dataset is first being created right now.
+        meta_tree = self.meta_tree if self.tree is not None else ()
 
-        meta_tree = self.meta_tree
+        conflicts = False
```

Comment:
would prefer to use `has_`/`is_`/`can_`/etc as a prefix for boolean variable names

Reply:
Done